modules/enterprise/server/jar/src/main/java/org/rhq/enterprise/server/content/ContentManagerBean.java

```diff
@@ -626,6 +628,76 @@ public void completeDeployPackageRequest(DeployPackagesResponse response) {
             entityManager.persist(history);
             persistedRequest.addInstalledPackageHistory(history);
         }
+
+        // Clean old version bits for packages..
+
+        for (DeployIndividualPackageResponse singleResponse : response.getPackageResponses()) {
+            PackageDetailsKey key = singleResponse.getKey();
+            Query packageVersionQuery = entityManager
+                    .createNamedQuery(PackageVersion.QUERY_FIND_BY_PACKAGE_DETAILS_KEY_WITH_NON_NULL_RESOURCE_TYPE);
+            packageVersionQuery.setParameter("packageName", key.getName());
+            packageVersionQuery.setParameter("packageTypeName", key.getPackageTypeName());
+            packageVersionQuery.setParameter("architectureName", key.getArchitectureName());
+            packageVersionQuery.setParameter("version", key.getVersion());
+            packageVersionQuery.setParameter("resourceTypeId", resourceTypeId);
+            PackageVersion packageVersion = (PackageVersion) packageVersionQuery.getSingleResult();
+            purgePackageBits(packageVersion.getGeneralPackage().getId());
+        }
+
+        deleteOrphanedBlobs();
+    }
+
+    public void removeHistoryDeploymentsBits(){
```

Comment:
Shouldn't all public API calls contain JavaDoc? Otherwise, this will be one more method that shows up in our product documentation without any information on the purpose or how it works.